### PR TITLE
misc: remove some deprecated methods in xDSL

### PR DESCRIPTION
**Pull request by superlopuh**
Removing some methods that have been deprecated for three weeks or more at @math-fehr's behest

### CODE REVIEW

File: xdsl/transforms/printf_to_llvm.py [hunk 1]
```diff
@@ -135,9 +135,7 @@ def match_and_rewrite(self, op: printf.PrintFormatOp, rewriter: PatternRewriter,
         rewriter.replace_matched_op(
             casts
             + [
-                ptr := llvm.AddressOfOp.get(
-                    globl.sym_name, llvm.LLVMPointerType.opaque()
-                ),
+                ptr := llvm.AddressOfOp(globl.sym_name, llvm.LLVMPointerType.opaque()),
```
Comment (webmiche) [inline xdsl/transforms/printf_to_llvm.py:138]: I guess this warning was just ignored? :thinking: 
Should we check that this does not happen?

**Review by webmiche — APPROVED** [review]
The LORD's work!

**Review by math-fehr — APPROVED** [review]

File: xdsl/transforms/printf_to_llvm.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/transforms/printf_to_llvm.py:138]: Yeah it was just ignored, I'm not sure exactly what the deal is, I think deprecated doesn't work well with static methods.

Comment (webmiche) [reply xdsl/transforms/printf_to_llvm.py:138]: Good thing we have used it for all the constructors xD

Comment (superlopuh) [reply xdsl/transforms/printf_to_llvm.py:138]: :)

**Pull request merged**